hal/halx86/generic/x86bios.c

```diff
@@ -123,6 +123,27 @@ ValidatePort(
 
         // CHECKME!
         case 0x3B6: return (Size <= 2);
+
+//
+// INVESTIGATE: It is known (Geoff Chappell) that the x86 BIOS emulator
+// also handles those ports separately. Check whether we can find some
+// video hardware ROMs that use them...
+//
+        case 0x70: case 0x71:
+            DPRINT1("*** Accessing CMOS port 0x%x, size %d, write %d\n", Port, Size, IsWrite);
+            return TRUE;
+
+        case 0xB1: case 0xB2:
+            DPRINT1("*** Accessing port 0x%x, size %d, write %d\n", Port, Size, IsWrite);
```

Comment:
Do this for `0x3B6` above too?

Reply:
0x3B6 is checked for by the VBox INT 10h VBIOS code handler when changing display resolution in a VBE mode.
Those other DPRINTs for the ports below are just for diagnosing those Geoff Chappell mentions in one of his articles.

Reply:
I'll ask Timo.

Reply:
0x3B6 came up in some debug prints (maybe KVM). So it is about the port number and how it's used. The return expression is based on that.